data/core/macros/utils.cfg

```diff
@@ -535,6 +535,54 @@
             [/and]
         [/filter_condition]
 
+        [store_unit]
+            [filter]
+                [filter_adjacent]
+                    x,y=$x1,$y1
+                [/filter_adjacent]
+                [not]
+                    x,y=$x2,$y2
+                [/not]
+                [not]
+                    x,y=$x1,$y1
+                [/not]
+            [/filter]
+            variable=force_remove
+        [/store_unit]
+
+        [foreach]
+            array=force_remove
+            [do]
+                [object]
+                    id=forced_re
+                    [filter]
+                        id=$force_remove[$i].id
+                    [/filter]
+                    silent=yes
+                    duration=turn end
```

Comment:
What if another unit attacks during the same turn? I haven't tested this, just wondering if this is a possible bug:

* Alice attacks Bob
* Charlie attacks Bob, but lingering effects from Alice's attack change Charlie's chance to hit.

Reply:
But objectif rénover in attack end event

Reply:
Thanks, I missed that detail.